### DISCUSSION TITLE
Lesson about Taxon Constraints

### DISCUSSION THREAD

**Discussion by sabrinatoro** [jamesaoverton/obook#328] [Tutorial requests]
**Tutorial Request:**

It would be useful to have a lesson about taxon constraints: how they work (e.g 'in taxon' is an object property, versus 'present/never in taxon' is an annotation property), how they are used,... 
Taxon constraint is something much more complicated than it looks and I think many curators do not appreciate this complexity. 
Note: this tutorial might be for advanced students, and not beginners.

**Suggested materials**

**Your nano-attribution (ORCID)**
https://orcid.org/0000-0002-4142-7153

**Comment by balhoff** [comment]
@sabrinatoro can you assign this to me? (I don't have permission). What type of article would this be? (tutorial, how-to, reference, explanation)  @nicolevasilevsky asked me to contribute this.

**Comment by nicolevasilevsky** [comment]
I assigned you, I'll give you access to this repo too.

Maybe it would be best as an explanation?
Actually, looks like we have this:
https://oboacademy.github.io/obook/explanation/taxon-constraints-explainer/

You could expand on this?

cc @matentzn

**Comment by sabrinatoro** [comment]
Thank you @balhoff (and @nicolevasilevsky )
When adding taxon constraints in Mondo, I expected that I could use these to assist in the classification (e.g. cat disease = diseases that are present in cats + diseases that are only on cats). My understanding is that "present in taxon" are annotations (when only in taxon could be subclass), but somehow the reasoner (though not in protege) is able to use these annotations for checks - but not classification (?). 
Nico (and the document) explained that these annotations are used as a shortcut that represents complex descriptions. 

Therefore, from my perspective, what is missing from the documentation above is the more practical aspect: 
- why can we not use constraints for classifications? 
- how can we check whether the taxon constraints create errors (ie if protege doesn't catch them, how can we check for them?). 
Thanks!

**Comment by matentzn** [comment]
@sabrinatoro what you are describing here is very very specific and very serious question for QC (also wrt to @balhoff taxon constraint plugin for protege).. but this is a piece of unsorted SOP/tech, way too far in the weeds  for a general OBO academy tutorial IMO. I would say the tutorial should cover:

1. What are taxon constraints and what kinds of constraints are there
1. How do we curate taxon constraints (annotation props, axioms etc)
1. How do we materialise logical constraints (robot expand, etc)
1. How do we use the constraints (display, species-specific subsets, QC, classification etc)

What you ar asking in the two bullet points is more like a bug report on ODK repo :D

**Comment by shawntanzk** [comment]
@sabrinatoro sorry I didnt see this till now (probably should follow this repo >.<) 
I've just done some minor updates on docs that might help (honestly thought this was in docs already, but realise it wasn't) - https://github.com/OBOAcademy/obook/pull/283 
I also have a presentation David and I did for the EBI ontologies team on this topic - https://docs.google.com/presentation/d/1uoQ53URkeoNf4Uroc_fUiExsJItj6MySKMIqX1KQyAc/edit?usp=sharing 
Hope that helps a bit :)

**Comment by sabrinatoro** [comment]
Thank you @shawntanzk

**Comment by nicolevasilevsky** [comment]
thanks @shawntanzk!

**Comment by matentzn** [comment]
This has been done by @balhoff https://oboacademy.github.io/obook/explanation/taxon-constraints-explainer/
